### PR TITLE
Add additional checks to ACLProvider to help troubleshoot

### PR DESCRIPTION
**Pull request by joeferner**
- [x] @srfarley
- [x] @mwizeman @dsingley @EvanOxfeld 
- [x] @joeybrk372 @sfeng88 @rygim @jharwig

### CODE REVIEW

File: core/core/src/main/java/org/visallo/core/security/ACLProvider.java [hunk 1]
```diff
@@ -70,13 +76,15 @@ public final ClientApiElementAcl elementACL(Element element, User user, Ontology
...
                 Concept concept = ontologyRepository.getConceptByIRI(iri);
+                checkNotNull(concept, "Could not find concept with iri: " + iri);
                 populatePropertyAcls(concept, element, user, propertyAcls);
                 iri = concept.getParentConceptIRI();
             }
         } else if (element instanceof Edge) {
             String iri = ((Edge) element).getLabel();
             while (iri != null) {
                 Relationship relationship = ontologyRepository.getRelationshipByIRI(iri);
+                checkNotNull(relationship, "Could not find concept with iri: " + iri);
```
Comment (joeybrk372) [inline core/core/src/main/java/org/visallo/core/security/ACLProvider.java]: ~~concept~~ relationship?

File: core/core/src/main/java/org/visallo/core/security/ACLProvider.java [hunk 2]
```diff
@@ -204,14 +212,22 @@ private ClientApiPropertyAcl newClientApiPropertyAcl(Element element, String key
     }
 
     private Element findElement(ClientApiElement apiElement) {
+        Element element;
         Authorizations authorizations = userRepository.getAuthorizations(userRepository.getSystemUser());
         if (apiElement instanceof ClientApiVertex) {
-            return graph.getVertex(apiElement.getId(), authorizations);
+            element = graph.getVertex(apiElement.getId(), authorizations);
+            if (element == null) {
```
Comment (srfarley) [inline core/core/src/main/java/org/visallo/core/security/ACLProvider.java]: Why not use `checkNotNull()` here like above?

File: core/core/src/main/java/org/visallo/core/security/ACLProvider.java [hunk 3]
```diff
@@ -204,14 +212,22 @@ private ClientApiPropertyAcl newClientApiPropertyAcl(Element element, String key
...
+                throw new VisalloException("could not find vertex with id: " + apiElement.getId());
+            }
         } else if (apiElement instanceof ClientApiEdge) {
-            return graph.getEdge(apiElement.getId(), authorizations);
+            element = graph.getEdge(apiElement.getId(), authorizations);
+            if (element == null) {
```
Comment (srfarley) [inline core/core/src/main/java/org/visallo/core/security/ACLProvider.java]: ... and here.

File: core/core/src/main/java/org/visallo/core/security/ACLProvider.java (hunk 1 above)
Comment (joeferner) [inline core/core/src/main/java/org/visallo/core/security/ACLProvider.java]: fixed

File: core/core/src/main/java/org/visallo/core/security/ACLProvider.java (hunk 2 above)
Comment (joeferner) [inline core/core/src/main/java/org/visallo/core/security/ACLProvider.java]: fixed

File: core/core/src/main/java/org/visallo/core/security/ACLProvider.java (hunk 3 above)
Comment (joeferner) [inline core/core/src/main/java/org/visallo/core/security/ACLProvider.java]: fixed

**Pull request merged**